### DISCUSSION TITLE
Duplicated request in SSG mode

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [johannschopplich/nuxt-kql#22] [Q&A]
Hoping someone can help with this strange behaviour I am seeing.
The pages and post show great during dev and all, but i just did a static site generate and I noticed some strange behaviour.

If I load the page directly, that is a post from kirby, it loads fine. As the KQL request was done server side, and data passed to the client.

If I navigate to a page, via a NuxtLink, I can see a post request being made in the browser console, which fails.
The post request  contains the kql query to kirby. It is request to:
`SITEURL/api/__kirby__/%24kqlnjn1BDTBxP`
And thus the page fails.

Now, even stranger.... if I load the post page directly, then navigate away from the page ( via NuxtLink ) and then go back to the same post page ( via NuxtLink ) it works fine and doesn't do the post request which fails.

Is there I way to make it truly static and not need to do any kirby data requests on the client side?

**Comment by johannschopplich** [comment]
If you use `useKql` and haven't disabled [`payloadExtraction`](https://nuxt.com/docs/guide/going-further/experimental-features#payloadextraction), all KQL queries will be written as a separate payload on `nuxi generate`. This payload will then be used on page load, even in the client.

Maybe that specific route wasn't prerendered at all? Can you check your SSG build if it includes the payload for the page where it seems to be missing? A reproduction is happily welcome. You can also book consulting from me if you continue to experience issues.

**Reply by pixelpaulaus** [reply]
i found the issue.  If i have any uri query params on the NuxtLink such as `?ref=blah` then it does the query again to KQL. But if I don't, it works fine. strange.

**Reply by johannschopplich** [reply]
But this should only the case of the query parameter is part of the `useKql` key. If you use `route.path`, this shouldn't be the case.
